src/components/pinned-header-footer-card/index.cjsx

```diff
@@ -111,17 +116,44 @@ module.exports = React.createClass
     window.scroll(0, @props.buffer + @state.offset)
     @setState(shouldBeShy: true)
 
+  getHeaderHeight: ->
+    header = @refs.header?.getDOMNode()
+    headerHeight = header?.offsetHeight or 0
+
+  setOriginalContainerMargin: ->
+    container = @refs.container?.getDOMNode()
+    return unless container
+
+    @setState(containerMarginTop: window.getComputedStyle(container).marginTop) if window.getComputedStyle?
+
+  setContainerMargin: ->
+    headerHeight = @getHeaderHeight()
+    container = @refs.container?.getDOMNode()
+    return unless container
+
+    @setState(headerHeight: headerHeight)
+    if @state.pinned
+      container.style.marginTop = (headerHeight + @props.containerBuffer) + 'px'
+    else
+      container.style.marginTop = @state.containerMarginTop
+
   componentDidMount: ->
     @setOffset()
     @updatePinState(0)
+    @setOriginalContainerMargin()
+    @setContainerMargin()
+
+    window.addEventListener('resize', @setContainerMargin)
```

Comment:
The exercise builder has a similar `throttle` method; maybe these can be consolidated?